frontends/concrete-python/examples/levenshtein_distance/levenshtein_distance.py

```diff
@@ -0,0 +1,424 @@
+            )
+            for _ in range(1000)
+        ]
+        inputset_mix = [
+            (
+                numpy.random.randint(2),
+                numpy.random.randint(args.max_string_length),
+                numpy.random.randint(args.max_string_length),
+                numpy.random.randint(args.max_string_length),
+                numpy.random.randint(args.max_string_length),
+            )
+            for _ in range(100)
+        ]
+
+        self.my_module = LevenshsteinModule.compile(
+            {
+                "equal": inputset_equal,
+                "mix": inputset_mix,
+                "constant": [i for i in range(len(self.mapping_to_int))],
+            },
+            show_mlir=args.show_mlir,
+            p_error=10**-20,
+            show_optimizer=args.show_optimizer,
+            comparison_strategy_preference=fhe.ComparisonStrategy.ONE_TLU_PROMOTED,
+            min_max_strategy_preference=fhe.MinMaxStrategy.ONE_TLU_PROMOTED,
+        )
+
+    def compute_in_simulation(self, list_patterns):
+        """Check equality between distance in simulation and clear distance."""
+        print("Computations in simulation\n")
+
+        for a, b in list_patterns:
+
+            print(f"    Computing Levenshtein between strings '{a}' and '{b}'", end="")
+
+            a_as_int = self.encode_string(a)
+            b_as_int = self.encode_string(b)
+
+            l1_simulate = levenshtein_simulate(self.my_module, a_as_int, b_as_int)
+            l1_clear = levenshtein_clear(a_as_int, b_as_int)
+
+            assert l1_simulate == l1_clear, f"    {l1_simulate=} and {l1_clear=} are different"
+            print(" - OK")
+
+    def compute_in_fhe(self, list_patterns, verbose=True, show_distance=False):
+        """Check equality between distance in FHE and clear distance."""
+        self.my_module.keygen()
+
+        # Checks in FHE
+        if verbose:
+            print("\nComputations in FHE\n")
+
+        for a, b in list_patterns:
+
+            print(f"    Computing Levenshtein between strings '{a}' and '{b}'", end="")
+
+            a_enc, b_enc = self.encode_and_encrypt_strings(a, b)
+
+            time_begin = time.time()
+            l1_fhe_enc = levenshtein_fhe(self.my_module, a_enc, b_enc)
+            time_end = time.time()
+
+            l1_fhe = self.my_module.mix.decrypt(l1_fhe_enc)
+
+            l1_clear = levenshtein_clear(a, b)
+
+            assert l1_fhe == l1_clear, f"    {l1_fhe=} and {l1_clear=} are different"
+
+            if not show_distance:
+                print(f" - OK in {time_end - time_begin:.2f} seconds")
+            else:
+                print(f" - distance is {l1_fhe}, computed in {time_end - time_begin:.2f} seconds")
+
+
+# Module FHE
+@fhe.module()
+class LevenshsteinModule:
+    @fhe.function({"x": "encrypted", "y": "encrypted"})
+    def equal(x, y):
+        """Assert equality between two chars of the alphabet."""
+        return x == y
+
+    @fhe.function({"x": "clear"})
```

Comment:
your code @BourgerieQuentin , sorry I had to copy paste, too much conflicts to manage

Reply:
by the way, it would be easier if the "constant" system was directly in CP, it's planned, right? CC @umut-sahin @BourgerieQuentin

Reply:
Yes, we should provide an API for that.

Reply:
I did https://github.com/zama-ai/concrete-internal/issues/810, maybe it's a dup